src/main/java/org/truffleruby/core/rope/RopeNodes.java

```diff
@@ -1139,15 +1139,6 @@ public boolean ropesEqual(Rope a, Rope b,
             return a.getEncoding() == b.getEncoding() && a.byteLength() == b.byteLength() && a.hashesMatch(b) && Arrays.equals(aBytes.execute(a), bBytes.execute(b));
         }
 
-        @Specialization(guards = "!isRope(b)")
-        public boolean ropeEqualNonRope(Rope a, Object b) {
```

Comment:
This node is only used via its execute method which takes two Ropes, so there simply cannot be something else than a rope. Moreover, the impl of isRope() was wrong.